kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/auth/AuthRepository.java

```diff
@@ -55,45 +56,26 @@ public void setSignInData(@NonNull SignInData signInData) {
 
     @Override
     public ObservableData<String> getAppID() {
+        loadCachedAppIDIfNeeded();
         return appId;
     }
 
-    @Override
-    public void getAuthToken(@NonNull final Callback<AuthToken> callback) {
-        if (cachedAuthToken != null) {
-            callback.onResponse(cachedAuthToken);
-        } else {
-            localData.getAuthToken(new Callback<AuthToken>() {
+    private void loadCachedAppIDIfNeeded() {
+        if (TextUtils.isEmpty(appId.getValue())) {
+            localData.getAppId(new Callback<String>() {
                 @Override
-                public void onResponse(final AuthToken response) {
-                    setAuthToken(response);
-                    callback.onResponse(cachedAuthToken);
-
+                public void onResponse(String appID) {
+                  postAppID(appID);
                 }
 
                 @Override
                 public void onFailure(Throwable t) {
-                    getRemoteAuthToken(callback);
+                    // No Data Available
```

Comment:
is this state reachable? see my other related comment

Reply:
it is reachable on the first run, where the appID is null but we don't need to do anything on that particular failure.